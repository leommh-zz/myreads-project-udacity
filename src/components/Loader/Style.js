const style = {
    main: {
        flex: 1
    }, 
    loading: {
        color: '#333'
    }
}

export default style;