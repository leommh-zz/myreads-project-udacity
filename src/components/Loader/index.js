import React from 'react';
import {CircularProgress, Grid} from '@material-ui/core';
import style from './style';

const Loader = () => (
    <Grid
        item 
        container   
        direction="row"
        justify="center"
        alignItems="center"
        style={style.main}
    >  
        <Grid item>
            <CircularProgress style={style.loading} thickness={7} />
        </Grid>
    </Grid>  
)

export default Loader;