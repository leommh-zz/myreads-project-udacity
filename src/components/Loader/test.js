import React from 'react';
import { shallow } from 'enzyme';
import Loader from './index';

describe('[Component] Loader', () => {
  
    it('Display Loader Component', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper).toMatchSnapshot();
    });
    
});