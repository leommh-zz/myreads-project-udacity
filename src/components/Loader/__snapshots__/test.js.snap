// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Loader Display Loader Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": <WithStyles(Grid)
        item={true}
>
        <WithStyles(CircularProgress)
                style={
                        Object {
                                "color": "#333",
                              }
                }
                thickness={7}
        />
</WithStyles(Grid)>,
      "container": true,
      "direction": "row",
      "item": true,
      "justify": "center",
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CircularProgress)
          style={
                    Object {
                              "color": "#333",
                            }
          }
          thickness={7}
/>,
        "item": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "color": "#333",
          },
          "thickness": 7,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": <WithStyles(Grid)
          item={true}
>
          <WithStyles(CircularProgress)
                    style={
                              Object {
                                        "color": "#333",
                                      }
                    }
                    thickness={7}
          />
</WithStyles(Grid)>,
        "container": true,
        "direction": "row",
        "item": true,
        "justify": "center",
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(CircularProgress)
            style={
                        Object {
                                    "color": "#333",
                                  }
            }
            thickness={7}
/>,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "color": "#333",
            },
            "thickness": 7,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
